sdk/src/trace/span.cc

```diff
@@ -77,16 +78,21 @@ Span::Span(std::shared_ptr<Tracer> &&tracer,
   }
   recordable_->SetName(name);
 
+  trace_api::TraceId trace_id;
+
   if (parent_span_context.IsValid())
   {
-    recordable_->SetIds(parent_span_context.trace_id(), GenerateRandomSpanId(),
-                        parent_span_context.span_id());
+    trace_id = parent_span_context.trace_id();
+    recordable_->SetIds(trace_id, GenerateRandomSpanId(), parent_span_context.span_id());
   }
   else
   {
-    recordable_->SetIds(GenerateRandomTraceId(), GenerateRandomSpanId(), trace_api::SpanId());
+    trace_id = GenerateRandomTraceId();
+    recordable_->SetIds(trace_id, GenerateRandomSpanId(), trace_api::SpanId());
   }
-  // TODO: Create and populate SpanContext for this span when SpanContext is fully implemented
+
+  span_context_ = std::unique_ptr<trace_api::SpanContext>(
+      new trace_api::SpanContext(trace_id, GenerateRandomSpanId(), trace_api::TraceFlags(), false));
```

Comment:
Why is this a different[ly random] span ID from the one on L86 or L91?

Reply:
It shouldn't have been. Fixed now.